freva-rest/src/freva_rest/databrowser_api/endpoints.py

```diff
@@ -261,21 +266,55 @@ async def stac_collection(
         translate=translate,
         **SolrSchema.process_parameters(request),
     )
-    status_code, total_count = await solr_search.validate_stac()
-    await solr_search.store_results(total_count, status_code)
+    status_code, total_count = await stac_instance.validate_stac()
+    await stac_instance.store_results(total_count, status_code)
     if total_count == 0:
         raise HTTPException(status_code=404, detail="No results found.")
     if total_count > max_results and max_results > 0:
         raise HTTPException(status_code=413, detail="Result stream too big.")
-    collection_id = f"freva-{str(uuid.uuid4())}"
-    background_tasks.add_task(solr_search.init_stac_collection, request, collection_id)
-    return {
-        "status": (
-            f"STAC catalog creation initiated successfully. "
-            f"To see the collection, use this link: "
-            f"{request.base_url}/stac/{collection_id}"
+
+    collection_id = f"Dataset-{(f'{flavour}-{str(uuid.uuid4())}')[:18]}"
+    if stac_dynamic:
```

Comment:
For my taste this if block is too big. Could you add the code that is withing this if block into a function? Or at least the `run_stac_creation` function.